internal/http/client.go

```diff
@@ -26,3 +39,39 @@ func (client Client) AppendDefaultHeaders(headers RequestHeaders) RequestHeaders
 
 	return headers
 }
+
+// Do does a request
+func (client Client) Do(method string, url string, headers RequestHeaders, body io.Reader) ([]byte, ResponseHeaders, error) {
+	if client.limiter != nil {
+		ctx := context.Background()
+		client.limiter.Wait(ctx)
+	}
+
+	headers = client.AppendDefaultHeaders(headers)
+
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return nil, ResponseHeaders{}, err
+	}
+
+	for _, header := range headers {
+		req.Header.Set(header.Name, header.Value)
+	}
+
+	resp, err := client.Client.Do(req)
+	if err != nil {
+		return nil, ResponseHeaders{}, err
+	}
+
+	err = HandleStatus(resp.StatusCode)
+	if err != nil {
+		return nil, ResponseHeaders{}, err
```

Comment:
I am wondering if this is dangerous as we are essentially swallowing an error, Shopifies HTTP error response could be more useful and meaningful to digest further up the chain. I would also be tempted to log anything from a known bad status.

Reply:
couldn't agree more, I've got an issue open to deal with it when we have time in future:
https://github.com/MOHC-LTD/httpshopify/issues/3

Reply:
I think in this case I would change it so we call the handle status method and return the provided response header and body even a simple println would be useful at this stage over what it currently does.